xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp

```diff
@@ -1681,6 +1670,24 @@ CDemuxStream* CDVDDemuxFFmpeg::AddStream(int streamIdx)
             st->hdr_type = StreamHdrType::HDR_TYPE_HDR10;
         }
 
+        uint8_t* dv_side_data = nullptr;
```

Comment:
This code seems to get a bit chaotic as now not used previous defined `side_data` variable but this variable is used again  in line 1691. I propose simplifícate a bit and move this block after line 1691:

```c++
        if (av_stream_get_side_data(pStream, AV_PKT_DATA_DOVI_CONF, nullptr) // DoVi
          st->hdr_type = StreamHdrType::HDR_TYPE_DOLBYVISION;
        else if (av_stream_get_side_data(pStream, AV_PKT_DATA_DYNAMIC_HDR10_PLUS, nullptr) // HDR10+
          st->hdr_type = StreamHdrType::HDR_TYPE_HDR10PLUS;
        else if (st->colorPrimaries == AVCOL_PRI_BT2020)
        {
          if (st->colorTransferCharacteristic == AVCOL_TRC_SMPTE2084) // HDR10
            st->hdr_type = StreamHdrType::HDR_TYPE_HDR10;
          else if (st->colorTransferCharacteristic == AVCOL_TRC_ARIB_STD_B67) // HLG
            st->hdr_type = StreamHdrType::HDR_TYPE_HLG;
        }
```

Side data is not used (only detect presence), so maybe is not need define new variables.

Reply:
Sure, I just basically extended the existing code

Reply:
I am not sure how this will affect this, but there's also the `size` variable, I'm not sure how if it is relevant?

Reply:
Optimized the code, can't test it yet though.

Reply:
I'm sorry, I think you misunderstood me: I meant to put the code that I have written just before:

`AVDictionaryEntry* rtag = av_dict_get(pStream->metadata, "rotate", NULL, 0);`


Current code is fine:

```c++
        int size = 0;
        uint8_t* side_data = nullptr;

        side_data = av_stream_get_side_data(pStream, AV_PKT_DATA_MASTERING_DISPLAY_METADATA, &size);
        if (side_data && size)
        {
          st->masteringMetaData = std::make_shared<AVMasteringDisplayMetadata>(
              *reinterpret_cast<AVMasteringDisplayMetadata*>(side_data));
          // file could be SMPTE2086 which FFmpeg currently returns as unknown so use the presence
          // of static metadata to detect it
          if (st->masteringMetaData->has_primaries && st->masteringMetaData->has_luminance)
            st->hdr_type = StreamHdrType::HDR_TYPE_HDR10;
        }

        side_data = av_stream_get_side_data(pStream, AV_PKT_DATA_CONTENT_LIGHT_LEVEL, &size);
        if (side_data && size)
        {
          st->contentLightMetaData = std::make_shared<AVContentLightMetadata>(
              *reinterpret_cast<AVContentLightMetadata*>(side_data));
        }

 // ------->  move new code here

       AVDictionaryEntry* rtag = av_dict_get(pStream->metadata, "rotate", NULL, 0);
```